service-base/src/main/java/org/eclipse/hono/service/resourcelimits/PrometheusBasedResourceLimitChecks.java

```diff
@@ -151,20 +187,33 @@ public PrometheusBasedResourceLimitChecks(final WebClient webClient,
         } else {
             final String queryParams = String.format("sum(%s{tenant=\"%s\"})", CONNECTIONS_METRIC_NAME,
                     tenant.getTenantId());
-            return executeQuery(queryParams)
+            return executeQuery(queryParams, span)
                     .map(currentConnections -> {
                         if (currentConnections < maxConnections) {
+                            span.finish();
                             return Boolean.FALSE;
                         } else {
+                            TracingHelper.logError(span, String.format(
+                                    "connection limit exceeded [tenant: %s, current connections: %s, max-connections: %s]",
+                                    tenant.getTenantId(), currentConnections, maxConnections));
                             log.trace(
                                     "connection limit exceeded [tenant: {}, current connections: {}, max-connections: {}]",
                                     tenant.getTenantId(), currentConnections, maxConnections);
+                            span.finish();
                             return Boolean.TRUE;
                         }
-                    }).otherwise(Boolean.FALSE);
+                    }).otherwise(failure -> {
+                        span.finish();
+                        return Boolean.FALSE;
+                    });
```

Comment:
IMHO you can simply finish the span here:
```
.otherwise(failure -> {
  return Boolean.FALSE;
})
.map(result -> {
  span.finish();
  return result;
});
```
instead of finishing the span in three places ...

Reply:
Now it has been replaced with map to finish the span.